### DISCUSSION TITLE
关于 java 调用的内存

### DISCUSSION THREAD

**Discussion by poplar-at-twilight** [PBH-BTN/PeerBanHelper#336] [讨论]
旧版本使用的内存是 `-Xmx256M`，6.0 版本以后，启动脚本是 `-Xmx386M`。

这似乎让 Java 的内存占用无法忽视了：

![image](https://github.com/user-attachments/assets/c4746998-d132-40f6-82f9-15168bc1f475)

----

当前的 Windows 启动脚本具体如下：

```
@echo off
chcp 65001
title PeerBanHelper
start ./jre/bin/javaw.exe -Xmx386M -XX:+UseG1GC -XX:+UseStringDeduplication -XX:+ShrinkHeapInSteps -Dfile.encoding=UTF-8 -Dstdout.encoding=UTF-8 -Dstderr.encoding=UTF-8 -Dconsole.encoding=UTF-8 "-Dhttp.proxyHost=127.0.0.1" "-Dhttp.proxyPort=7890" "-Dhttps.proxyHost=127.0.0.1" "-Dhttps.proxyPort=7890" -jar PeerBanHelper.jar silent
```

**Comment by poplar-at-twilight** [comment]
![image](https://github.com/user-attachments/assets/4d93e7ab-c00f-4d9a-b4f7-b7e928590d3a)

重启了一下系统

**Comment by Einsfier** [comment]
刚升到6.0的时候，猛一看群晖统计的内存占用怎么300M+了。
但想想这么多功能，java是这样的，安啦，不上四位数我都能接受
![image](https://github.com/user-attachments/assets/3363f8e6-6552-4e23-b95b-932d950d25e4)

**Comment by Ghost-chu** [comment]
PBH最低可使用128M运行，386只是为了避免缓存击穿
